src/pages/AccountDeletion.js

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState, useContext } from "react";
+import { UserContext } from "../contexts/UserContext";
+import InfoPage from "./info_page_shared/InfoPage";
+import Header from "./info_page_shared/Header";
+import Heading from "./info_page_shared/Heading";
+import Main from "./info_page_shared/Main";
+import Footer from "./info_page_shared/Footer";
+import LoadingAnimation from "../components/LoadingAnimation";
+import redirect from "../utils/routing/routing";
+import SessionStorage from "../assorted/SessionStorage";
+import { APP_DOMAIN } from "../appConstants";
+
+const TIME_BEFORE_REDIRECT = 5000;
+
+export default function AccountDeletion({ api }) {
+  const user = useContext(UserContext);
+  const [headingMsg, setHeadingMsg] = useState("We are deleting your account");
+  const [errorOccurred, setErrorOccurred] = useState();
+  const [isAccountDeleted, setIsAccountDeleted] = useState();
+  useEffect(() => {
+    console.log(user);
+    if (SessionStorage.hasUserConfirmationForAccountDeletion()) {
+      setErrorOccurred(false);
+      setIsAccountDeleted(false);
+      SessionStorage.setUserConfirmationForAccountDeletion(false);
+      api.deleteUser(
+        () => {
+          setIsAccountDeleted(true);
+          setHeadingMsg("✅ Your account has been successfully deleted!");
+          setTimeout(() => {
+            user.logoutMethod();
+          }, [TIME_BEFORE_REDIRECT]);
```

Comment:
my linter complains about the `setTimeout` second parameter being an array. Should it not be a simple int?

Reply:
It's right. It somehow was still working, which is strange - but it is expected to be an int.